### PR TITLE
[DRAFT] Security feature SuperAdmin documentation updates 

### PR DESCRIPTION
**Pull request by leanneeliatra**
### Description
The SuperAdmin role was added and this will be added to the documentation. 

### Issues Resolved
**One part of**: [DOC] Missing documentation for security features #433. The part this PR addresses is: CRUD APIs for default configurations. (Default roles, users and role-mapping can be updated by super admin now).

The original PR for the work of this ticket can be found here: [Added SuperAdmin check to allow update/delete/add of reserved config](https://github.com/opensearch-project/security/pull/242), this PR is concerned with adding the documentation to support the SuperAdmin addition.

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by leanneeliatra** [comment]
Hi @Naarcha-AWS & @hdhalter. I opened this ticket as a draft to get some input from you please. I'm not sure if the file I have included these changes in, is the right place for this addition to the documentation.

Could you take a look and let me know please. I am adding a section to the documentation to convey the addition of the SuperAdmin role. The SuperAdmin can add, update and delete the reserved configuration such as roles, roles_mapping, internal_users, action_groups and tenants.

I have included these changes in the `_security/access-control/users-roles.md` file at the end, or on the `# Users and roles` page. Do you think this is a suitable place to add these changes? Thank you.

**Comment by leanneeliatra** [comment]
We reviewed these changes internally and I am currently in the process of understanding the original PR fully,  so I can update the documentation appropriately.

**Comment by leanneeliatra** [comment]
I reached out to Hardik Shah via slack to discuss the original PR in more depth.

File: _security/access-control/users-roles.md [hunk 1]
```diff
@@ -218,3 +218,40 @@ Map the role to your user:
...
+## SuperAdmin role
+
+The `superAdmin` role enables adding, updating, and deleting reserved configurations like roles, roles_mapping, internal_users, action_groups, and tenants. Previously, these configurations were loaded from default YAML files. Now, the `superAdmin` role now has the privilege to add, update, and delete by using API calls.
+
+### Example API calls:
+Following are examples of API calls for `superAdmin` actions:
+
+Adding a Reserved Configuration:
+```
+curl -X PUT https://localhost:9200/_opendistro/_security/api/roles/new_role -k -H 'Content-Type: application/json' -d '{
```
Comment (hardik-k-shah) [inline _security/access-control/users-roles.md:231]: also don't user need to pass Admin certificate for invoking this API, other than invoking locally from the node? 
without identity to this API, system won't know the caller is super admin. And identity for super admin was being passed as client certificate with dn name which is allowlisted in config.yml 

Can you verify working these?

**Comment by leanneeliatra** [comment]
Thank you @hardik-k-shah for your comments much appreciated. Still testing these changes locally to ensure correct documentation of the SuperAdmin.

**Comment by hdhalter** [comment]
@leanneeliatra - Do you want to close this PR in favor of https://github.com/opensearch-project/documentation-website/pull/7069?

**Comment by leanneeliatra** [comment]
> @leanneeliatra - Do you want to close this PR in favor of #7069?

Hi @hdhalter, they are similar tickets. 
I think given that this ticket originally pertains to a lot of code changes, and comments. It might still be worth it to review this on it's own, following the closure of https://github.com/opensearch-project/documentation-website/pull/7069. Thank you though. For now if it's okay I would vote to keep them both open.

**Comment by leanneeliatra** [comment]
> @leanneeliatra - Do you want to close this PR in favor of #7069?

Hi @hdhalter yes please. I can ensure all aspects are covered in 7069.

**Comment by hdhalter** [comment]
Closing in favor of https://github.com/opensearch-project/documentation-website/pull/7069

**Pull request closed without merging**